ddtrace/_trace/tracer.py

```diff
@@ -230,7 +228,7 @@ def __init__(
         self.enabled = config._tracing_enabled
         self.context_provider = context_provider or DefaultContextProvider()
         # _user_sampler is the backup in case we need to revert from remote config to local
-        self._user_sampler: Optional[BaseSampler] = DatadogSampler()
+        self._user_sampler: Optional[DatadogSampler] = DatadogSampler()
```

Comment:
is this actually `Optional`? sounds like it is required.

Should this behavior be baked in somewhere rather than requiring two `DatadogSampler`s? (for a follow-up, my immediate feedback is only on the typing here)

Reply:
I actually addressed this in a different PR. This should not be optional. Making it optional causes a lot of typing issues.

Reply:
https://github.com/DataDog/dd-trace-py/pull/12582/files#diff-e2ff2c401c4b927861c9fc104deb21aee510e2b273cc4569315bd611a64ff3ba